File: kotlin/kotlin-step2/java-app/src/main/java/com/group/libraryapp/LibraryAppApplication.java
package com.group.libraryapp;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;


/**
 * Kotlin 리팩토링 계획 세우기
 * 1) Domain
 * 2) Repository (Spring Bean이고, 다른 스프링 bean 의존성 X)
 * 3) Service (비즈니스 로직)
 * 4) Controller, DTO ()
 */
@SpringBootApplication
public class LibraryAppApplication {

  public static void main(String[] args) {
    SpringApplication.run(LibraryAppApplication.class, args);
  }

}

File: kotlin-app_v2/src/main/kotlin/com/group/libraryapp/LibraryAppApplication.kt
package com.group.libraryapp

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication

@SpringBootApplication
class LibraryAppApplication

// top line에 함수를 만드는 경우 static 함수처럼 간주된다.
fun main(args: Array<String>) {
    // 확장함수
    runApplication<LibraryAppApplication>(*args) // 가변인자 *args
}